### ISSUE TITLE
Error in sample function when creating ExoData

### ISSUE THREAD

**Issue by christianpoitras** [welch16/ChIPexoQual#10]
When creating an ExoData, this error can occur:
Error: cannot take a sample larger than the population when 'replace = FALSE'

This seems to be related to the number of reads in the file since reading the file using readGAlignments and sampling a high number of those reads often allow the creation of ExoData without the error.

**Comment by christianpoitras** [comment]
Here is an analysis with data that shows the error.
https://1drv.ms/u/s!ApMySalZ9HztheAgwyVcKMN-w27lxA?e=Dhsf7j

If the sampling variable at the beginning of the script is changed to a value equals to 4 millions or above, the error are more frequent. I haven't found the exact source of the error.

**Comment by welch16** [comment]
Hi @christianpoitras, I went through just script and I think here is the issue, when reads are sampled the pipeline partitions the coverage into less regions. The error is happening when the quality statistics distribution is being calculated, and that step basically samples some regions among all, and then does a linear model.

I added the default parameters, based in human / mouse datasets that I analyzed for the [ChIPexoQual manuscript](https://academic.oup.com/nar/article/45/15/e145/4056229#119461300). My guess is that your data was generated from a smaller genome, and you are not getting enough regions after partitioning the coverage. Perhaps you could tweak the `nregions` parameter in `ExoData` to avoid this error.

Thanks for the feedback, Rene

**Comment by christianpoitras** [comment]
Hi Rene,
My data is from Yeast.
I tried a few values for the `nregions` parameter without success. The documentation is not clear as to what value is recommended. What value did you use for E. coli?

**Comment by welch16** [comment]
I don't recall exactly, but perhaps 400 or 500 may work

**Comment by christianpoitras** [comment]
Hi Rene,

I modified my program to test the maximum value for nregions to allow all reads to be processed without getting an error. For the example dataset I provided, the maximum value is 55 regions. For another dataset where both replicates are more similar, the maximum value is 122 regions. In both cases, the number of regions is too small to produce reliable plots.

The plots seem better when I sample around 3 million reads from each samples and use a value around 500 regions. The generated plots are similar when the program is executed multiple times with a random sampling of 3 million reads.
Does it seem logical to you?

Thanks for the help!
